auth_options.go

```diff
@@ -81,6 +81,17 @@ type AuthOptions struct {
 	// TokenID allows users to authenticate (possibly as another user) with an
 	// authentication token ID.
 	TokenID string `json:"-"`
+
+	// Scope determines the scoping of the authentication request.
+	Scope *AuthScope `json:"-"`
+}
+
+// AuthScope allows a created token to be limited to a specific domain or project.
+type AuthScope struct {
```

Comment:
I wasn't sure to call this `Scope` or `AuthScope` -- the latter being more descriptive for the package as a whole. But I opted to call it `Scope` in the `AuthOptions` struct since "auth" is implied there. I can keep it this way, change both to `Scope`.